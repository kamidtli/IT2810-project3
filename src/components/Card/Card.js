import React from 'react';
import { connect } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Star from '@material-ui/icons/Stars';
import { Link } from 'react-router-dom';
import WatchlistButton from '../WatchlistButton/WatchlistButton';

const useStyles = makeStyles((theme) => ({
  card: {
    width: '100%',
    maxWidth: 345,
    display: 'flex',
    flexDirection: 'column',
  },
  media: {
    height: 150,
  },
  content: {
    textOverflow: 'ellipsis',
    marginBottom: '35px',
  },
  actions: {
    justifyContent: 'flex-end',
    alignItems: 'center',
  },
  link: {
    color: theme.palette.text.primary,
    textDecoration: 'none',
  },
  spacer: {
    flexGrow: 1,
  },
  rating: {
    margin: 0,
  },
  ratingIcon: {
    padding: theme.spacing(1),
  },
}));

/**
 * Card component for showing a movie card
 * @param {*} props
 * Props is a JSON object it gets from
 * GraphQL with data about
 * id, title, link, imgUrl, short description
 */

function MediaCard(props) {
  const classes = useStyles();

  return (
    <Card className={classes.card}>
      <Link
        to={`/movie/${props.id}`}
        className={`${classes.link} ${classes.spacer}`}
        onClick={() => (props.addCurrentId(props.id))}
      >
        <CardActionArea>
          <CardMedia
            className={classes.media}
            image={
              props.imgUrl
              || 'https://images.unsplash.com/photo-1485846234645-a62644f84728?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=1340&q=80'
            }
            title={props.title}
          />
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
              {props.title}
            </Typography>
            <Typography variant="body2" color="textSecondary" component="p">
              {props.shortDescription || 'Not available'}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Link>
      <CardActions className={classes.actions}>
        <Star className={classes.ratingIcon} />
        <Typography gutterBottom className={classes.rating}>
          {props.rating}
        </Typography>
        <div className={classes.spacer} />
        <WatchlistButton id={props.id} />
      </CardActions>
    </Card>
  );
}

// Empty because we don't need props here, but need the function in 'connect'
const mapStateToProps = (state) => ({
  currentId: state.currentCard,
});

const mapDispatchToProps = (dispatch) => ({
  addCurrentId: (id) => dispatch({ type: 'CURRENT_CARD', id }),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(MediaCard);
