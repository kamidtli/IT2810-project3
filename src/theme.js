import { createMuiTheme, responsiveFontSizes } from '@material-ui/core/styles';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#F6AE2D',
    },
    secondary: {
      main: '#000d34',
    },
  },
  status: {
    danger: 'orange',
  },
});

const responsiveTheme = responsiveFontSizes(theme);

export default responsiveTheme;
