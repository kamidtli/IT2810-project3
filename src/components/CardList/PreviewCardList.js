import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import { makeStyles } from '@material-ui/core/styles';
import { useQuery } from '@apollo/react-hooks';
import gql from 'graphql-tag';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import Card from '../Card/Card';
import search from '../../reducers/search';
import SortBar from '../SortBar/sortBar';
import FilterBar from '../FilterBar/filterBar';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    maxWidth: 1000,
  },
  gridItem: {
    display: 'flex',
    justifyContent: 'center',
    padding: theme.spacing(2),
  },

}));

export default function PreviewCardList() {
  const classes = useStyles();

  const SEARCH_QUERY = gql`
  {
    findMoviesBasedOnYearRange(min:2015, max: 2019, sort:"-released", pagination:6,){
        _id
        title
        plot
        poster
    }
  }
  `;


  const {
    data, loading, error, fetchMore,
  } = useQuery(SEARCH_QUERY);

  if (loading) return <p>LOADING</p>;
  if (error) return <p>{error.message}</p>;


  const renderData = (toRender) => (
    toRender.map((movie) => (
      <Grid item xs={12} md={6} lg={4} key={movie._id} className={classes.gridItem}>
        <Card
          title={movie.title}
          shortDescription={movie.plot}
          imgUrl={movie.poster}
          id={movie._id}
        />
      </Grid>
    ))
  );


  return (
    <div>
      <div className={classes.root}>
        <Grid container>
          {renderData(data.findMoviesBasedOnYearRange)}
        </Grid>
      </div>
    </div>
  );
}
