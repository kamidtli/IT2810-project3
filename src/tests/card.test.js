import React from 'react';
import { shallow } from 'enzyme';
import { MockedProvider } from '@apollo/react-testing';
import { ThemeProvider } from '@material-ui/styles';
import Card from '../components/Card/Card';
import PreviewCardList from '../components/CardList/PreviewCardList';
import theme from '../theme';

describe('Card component', () => {
  it('renders without crashing', () => {
    const card = shallow(<ThemeProvider theme={theme}><Card /></ThemeProvider>);
    expect(card.toJSON).toMatchSnapshot();
  });
});
