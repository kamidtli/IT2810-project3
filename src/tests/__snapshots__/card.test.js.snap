// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders without crashing 1`] = `undefined`;
